File: backend/app/models/agent.py
"""
Agent database model
"""
from datetime import datetime
from sqlalchemy import String, Text, Float, Integer, DateTime, ForeignKey, JSON
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.database import Base


class Agent(Base):
    """Agent model for managing AI agents"""
    __tablename__ = "agents"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    name: Mapped[str] = mapped_column(String(100), nullable=False, index=True)
    description: Mapped[str] = mapped_column(Text, nullable=True)
    system_prompt: Mapped[str] = mapped_column(Text, nullable=False)
    
    # Model configuration
    model_provider: Mapped[str] = mapped_column(String(50), nullable=False, index=True)
    model_name: Mapped[str] = mapped_column(String(100), nullable=False)
    
    # Tools configuration (stored as JSON)
    tools: Mapped[str] = mapped_column(Text, nullable=True)  # JSON string
    
    # Model parameters
    temperature: Mapped[float] = mapped_column(Float, default=0.7, nullable=False)
    max_tokens: Mapped[int] = mapped_column(Integer, default=2000, nullable=False)
    
    # User relationship
    user_id: Mapped[int] = mapped_column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    
    # Timestamps
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at: Mapped[datetime] = mapped_column(
        DateTime, 
        default=datetime.utcnow, 
        onupdate=datetime.utcnow, 
        nullable=False
    )
    
    # Relationship to User
    user: Mapped["User"] = relationship("User", backref="agents")

    def __repr__(self) -> str:
        return f"<Agent(id={self.id}, name={self.name}, model={self.model_provider}/{self.model_name})>"

File: frontend/src/types/agent.ts
export interface Agent {
  id: string
  name: string
  description: string
  system_prompt: string
  model_name: string
  tools: string[]
  temperature?: number
  max_tokens?: number
  user_id: string
  created_at: string
  updated_at: string
}

export interface CreateAgentRequest {
  name: string
  description: string
  system_prompt: string
  model_name: string
  tools: string[]
  temperature?: number
  max_tokens?: number
}

export interface UpdateAgentRequest {
  name?: string
  description?: string
  system_prompt?: string
  model_name?: string
  tools?: string[]
  temperature?: number
  max_tokens?: number
}

export interface AgentListResponse {
  agents: Agent[]
  total: number
  page: number
  page_size: number
}
